src/OrleansRuntime/Catalog/Catalog.cs

```diff
@@ -1576,6 +1576,11 @@ private List<ActivationData> TryGetActivationDatas(List<ActivationAddress> addre
 
         public void SiloStatusChangeNotification(SiloAddress updatedSilo, SiloStatus status)
         {
+            this.ScheduleTask(() => this.OnSiloStatusChange(updatedSilo, status));
```

Comment:
This relies on implementers to know this is necessary.  Not obvious to maintainers or those writing new listeners.